src/java.base/share/classes/jdk/internal/lang/stable/StableValueElement.java

```diff
@@ -86,40 +96,69 @@ public String toString() {
         return StableUtil.toString(this);
     }
 
+    // Avoid creating lambdas
+    private static final Function<Supplier<?>, ?> SUPPLIER_EXTRACTOR =
+            new Function<Supplier<?>, Object>() {
+                @Override
+                public Object apply(Supplier<?> supplier) {
+                    return supplier.get();
+                }
+            };
+
+    @SuppressWarnings("unchecked")
+    private static <K, V> Function<? super K, ? extends V> supplierExtractor() {
+        return (Function<? super K, ? extends V>) SUPPLIER_EXTRACTOR;
+    }
+
     @Override
     public V computeIfUnset(Supplier<? extends V> supplier) {
-        return computeIfUnset0(supplier, Supplier::get);
+        // Todo: This creates a lambda
+        return computeIfUnsetShared(supplier, Supplier::get);
```

Comment:
```suggestion
        return computeIfUnsetShared(supplier, supplierExtractor());
```

Reply:
Yes. This is a work in progress. I will explore using a `BiFunction` instead.